cli/src/commands/migrateToInQueries.js

```diff
@@ -0,0 +1,269 @@
+import { fromJS, Map } from 'immutable';
+import lodash from 'lodash';
+import logger from 'lib/logger';
+import Dashboard from 'lib/models/dashboard';
+import StatementForwarding from 'lib/models/statementForwarding';
+import Visualisation from 'lib/models/visualisation';
+import User from 'lib/models/user';
+
+/**
+ * @param {immutable.List} criteriaPath
+ * @returns {boolean}
+ */
+const isActor = criteriaPath =>
+  criteriaPath.length === 2
+    && criteriaPath[0] === 'statement'
+    && criteriaPath[1] === 'actor';
+
+/**
+ * @param {immutable.List} criteriaPath
+ * @returns {boolean}
+ */
+const isPersonaId = criteriaPath =>
+  criteriaPath.length > 2
+    && criteriaPath[0] === 'persona'
+    && criteriaPath[1] === 'import';
+
+/**
+ * @param {immutable.List} criteriaPath
+ * @returns {boolean}
+ */
+const isContextInstructor = criteriaPath =>
+  criteriaPath.length === 3
+    && criteriaPath[0] === 'statement'
+    && criteriaPath[1] === 'context'
+    && criteriaPath[2] === 'instructor';
+
+/**
+ * @param {immutable.Map} condition
+ * @returns {boolean}
+ */
+const shouldConvert = (condition) => {
+  const comment = condition.get('$comment');
+  const criteriaPath = JSON.parse(comment).criteriaPath || [];
+
+  if (isActor(criteriaPath) || isPersonaId(criteriaPath) || isContextInstructor(criteriaPath)) {
+    return false;
+  }
+
+  return true;
+};
+
+/**
+ * Convert queries on immutable object layer
+ *
+ * @param {immutable.Map} oldQuery
+ * @returns {immutable.Map}
+ */
+const _buildNewQuery = (oldQuery) => {
```

Comment:
I can see that we've exported another function here called `buildNewQuery`, hence why you've put an underscore on it. Do you think there's a better name to give to these two functions so that we don't need the underscore?

Reply:
Pretty sure its just to denote its a private method (not exported) - not a big one for me

Reply:
Yeah just popped out to me, not that fussed either.